### PR TITLE
docs for i18n + missing features and tests

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

## 📖 Description

This PR adds the docs for i18n, some missing features from vCurrent, and some missing tests.

### CODE REVIEW

**Comment by zewa666** [comment]
Lgtm

File: docs/user-docs/2. app-basics/20. internationalization.md [hunk 1]
```diff
@@ -1,3 +1,915 @@
 # Internationalization
```
[the remark this answers is not in the document]
Comment (Sayan751) [reply docs/user-docs/2. app-basics/20. internationalization.md]: That seems more streamlined. I agree the recipes and build tool specific stuffs disturbs the flow. A link to the specific sections would be better. 

I think the guidelines are here: https://github.com/aurelia/aurelia/blob/master/docs/user-docs/6.%20community-contributions/5.%20writing-documentation.md
Missed it completely. I can make some minor corrections at least, such as remove the ToC, and move the migration guide to specific file, and move the recipe to examples dir. I only wonder how the cross-page links will work.

[the remark this answers is not in the document]
Comment (Sayan751) [reply docs/user-docs/2. app-basics/20. internationalization.md]: Okay. In that case, it is better to open a docs Issue to keep track of things. What do you think?

[the remark this answers is not in the document]
Comment (Sayan751) [reply docs/user-docs/2. app-basics/20. internationalization.md]: Would you please create that then, considering you have the broader overview on this topic. Difficult for me to flesh it out :(

**Pull request merged**